testutil/testnode/node_interaction_api.go

```diff
@@ -130,14 +134,37 @@ func (c *Context) PostData(account string, ns, msg []byte) (*sdk.TxResponse, err
 	if err != nil {
 		return nil, err
 	}
-
-	res, err := c.BroadcastTxCommit(rawTx)
+	var res *sdk.TxResponse
+	switch broadcastMode {
+	case flags.BroadcastSync:
+		res, err = c.BroadcastTxSync(rawTx)
+	case flags.BroadcastAsync:
+		res, err = c.BroadcastTxAsync(rawTx)
+	case flags.BroadcastBlock:
+		res, err = c.BroadcastTxCommit(rawTx)
+	default:
+		return nil, fmt.Errorf("unsupported return type %s; supported types: sync, async, block", c.BroadcastMode)
+	}
 	if err != nil {
 		return nil, err
 	}
 	if res.Code != abci.CodeTypeOK {
-		return nil, fmt.Errorf("failure to broadcast tx sync: %s", res.RawLog)
+		return res, fmt.Errorf("failure to broadcast tx sync: %s", res.RawLog)
 	}
 
 	return res, nil
 }
+
+// FillBlock creates and submits a single transaction that is large enough to
+// create a square of the desired size. broadcast mode indicates if the tx
+// should be submitted async, sync, or block. (see flags.BroadcastModeSync). If
```

Comment:
[question] should this be `flags.BroadcastMode` per https://github.com/cosmos/cosmos-sdk/blob/7781cdb3d20bc7ebac017452897ce1e6ab3903ef/client/flags/flags.go#L114

```suggestion
// should be submitted async, sync, or block (see flags.BroadcastMode). If
```

Note: `block` mode was removed in https://github.com/cosmos/cosmos-sdk/pull/12659. Since celestia-app currently uses cosmos-sdk v0.46.0 we can still use `block` here but note to future selves that we'll need to remove it when we upgrade cosmos-sdk to a release with that change

Reply:
yeah, blocking is just convenient atm and we will have to fix this in the future across the board.

Reply:
#868